src/sentry/incidents/logic.py

```diff
@@ -1556,15 +1627,31 @@ def _get_alert_rule_trigger_action_slack_channel_id(
     return channel_data.channel_id
 
 
-def _get_alert_rule_trigger_action_discord_channel_id(name: str, integration_id: int) -> str:
+def _get_alert_rule_trigger_action_discord_channel_id(
+    name: str | None, integration_id: int
+) -> str | None:
     from sentry.integrations.discord.utils.channel import validate_channel_id
 
     integration = integration_service.get_integration(integration_id=integration_id)
     if integration is None:
         raise InvalidTriggerActionError("Discord integration not found.")
     try:
+        # TODO(RyanSkonnord): Force this function's `name` param to be non-null.
+        #
+        # AlertRuleTriggerActionSerializerTest.test_discord_channel_id_none covers
+        # the case of `name` being None explicitly. However, from eyeballing
+        # validate_channel_id, it looks like this can only lead to an error. The unit
+        # test case originates in https://github.com/getsentry/sentry/pull/58005,
+        # where it looks like it was targeting a spurious failure mode that occurs
+        # while building the description string.
+        #
+        # I'd like to move the null check upstream, clean up the associated nullable
+        # function params, remove the piecemeal checks from
+        # _get_target_identifier_display_from_target_value, and modify
+        # test_discord_channel_id_none to assert for an explicit null check (or just
+        # delete it).
         validate_channel_id(
-            channel_id=name,
+            channel_id=name,  # type: ignore[arg-type]
```

Comment:
hmmm maybe split this out and do it first rather than following up later?

Reply:
I considered that but it's much easier to understand the motivation with the type annotations in place.

Here's a follow-up PR: https://github.com/getsentry/sentry/pull/76553. We could merge it into this PR or rebase it onto master after this one is merged; lmk if you have a preference between the two.

Reply:
Now merged into this one.